### PR TITLE
feat: pin node for renovate

### PR DESCRIPTION
**Pull request by viceice**
Use hardcoded node version for renovate executables

### CODE REVIEW

File: bin/docker-entrypoint.sh [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
+  . "$BASH_ENV"
 fi
 
 if [[ "${1:0:1}" = '-' ]]; then
   # assume $1 is renovate flag
   set -- renovate "$@"
 fi
 
-if [[ ! -x "$(command -v ${1})" ]]; then
+if [[ ! -x "$(command -v "${1}")" ]]; then
```
Comment (rarkins) [inline bin/docker-entrypoint.sh:13]: Double quoting works?

Comment (viceice) [reply bin/docker-entrypoint.sh:13]: yes. `$()` is a sub shell 🙃

File: Dockerfile [hunk 2]
```diff
@@ -29,6 +29,12 @@ RUN set -ex; \
   yarn build; \
   chmod +x dist/*.js;
 
+# hardcode node version to renovate
+RUN set -ex; \
+  NODE_VERSION=$(node -v | cut -c2-); \
```
Comment (rarkins) [inline Dockerfile:34]: Is NODE_VERSION likely to get used or confused anywhere else? Eg name it DEFAULT_NODE_VERSION instead?

File: Dockerfile [hunk 3]
```diff
@@ -29,6 +29,12 @@ RUN set -ex; \
...
+  sed -i "1 s:.*:#\!\/opt\/buildpack\/tools\/node\/${NODE_VERSION}\/bin\/node:" "dist/renovate.js"; \
+  sed -i "1 s:.*:#\!\/opt\/buildpack\/tools\/node\/${NODE_VERSION}\/bin\/node:" "dist/config-validator.js";
```
Comment (viceice) [inline Dockerfile:36]:
```suggestion
  DEFAULT_NODE_VERSION=$(node -v | cut -c2-); \
  sed -i "1 s:.*:#\!\/opt\/buildpack\/tools\/node\/${DEFAULT_NODE_VERSION}\/bin\/node:" "dist/renovate.js"; \
  sed -i "1 s:.*:#\!\/opt\/buildpack\/tools\/node\/${DEFAULT_NODE_VERSION}\/bin\/node:" "dist/config-validator.js";
```

Should we do this?

**Review by rarkins — APPROVED** [review]

**Pull request merged**